boto/exception.py

```diff
@@ -583,3 +583,18 @@ class InvalidInstanceMetadataError(Exception):
     def __init__(self, msg):
         final_msg = msg + '\n' + self.MSG
         super(InvalidInstanceMetadataError, self).__init__(final_msg)
+
+
+class BaseEndpointResolverError(Exception):
```

Comment:
Yeah I was thinking if we can consolidate these along with ``EndpointResolver`` in the same location so it is really easy to do backports. Having them in ``exceptions`` makes that slightly harder.